File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/article/domain/ArticleScreen.java
package pl.orbitemobile.wspolnoty.activities.article.domain;

import io.reactivex.Single;
import pl.orbitemobile.wspolnoty.data.RemoteRepository;
import pl.orbitemobile.wspolnoty.data.entities.Article;

public class ArticleScreen {
    
    private RemoteRepository mRemoteRepository;
    
    public ArticleScreen() {
        mRemoteRepository = new RemoteRepository();
    }
    
    public Single<Article> getRemoteArticleDetails(String articleDetailsUrl) {
        return mRemoteRepository.getArticleDetails(articleDetailsUrl);
    }
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/article/domain/ArticleScreen.kt
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.article.domain

import io.reactivex.Single
import pl.orbitemobile.kotlinrx.fromIoToMainThread
import pl.orbitemobile.wspolnoty.data.RemoteRepositoryImpl
import pl.orbitemobile.wspolnoty.data.dto.ArticleDTO

class ArticleScreen {

    private val mRemoteRepository: RemoteRepositoryImpl = RemoteRepositoryImpl.instance

    fun getRemoteArticleDetails(articleDetailsUrl: String): Single<ArticleDTO> =
            mRemoteRepository.getArticleDetails(articleDetailsUrl).fromIoToMainThread()
}